notifications/tests.py

```diff
@@ -56,3 +58,75 @@ def test_surge_translation(self):
                         getattr(surge_alert, build_localized_fieldname(field, lang)),
                         self.aws_translator._fake_translation(original_value, lang, 'en') if lang != 'en' else original_value,
                     )
+
+class SurgeAlertFilteringTest(APITestCase):
+
+    def test_filtering(self):
+        region_1 = Region.objects.create(name=1)
+        region_2 = Region.objects.create(name=2)
+        country_1 = Country.objects.create(name='Atlantis', iso3='ATL', region=region_1)
+        country_2 = Country.objects.create(name='Neptunus', iso3='NPT', region=region_2)
+        molnix_tag_1 = MolnixTag.objects.create(
+            molnix_id=3,
+            name='OP-6700',
+            description='Middle East Crisis | MENA',
+            tag_type='regular',
+            tag_category='molnix_operation')
+        molnix_tag_2 = MolnixTag.objects.create(
+            molnix_id=5,
+            name='L-FRA',
+            description='French',
+            tag_type='language',
+            tag_category='molnix_language')
+        molnix_tag_3 = MolnixTag.objects.create(
+            molnix_id=17,
+            name='AMER',
+            description='Americas Region',
+            tag_type='regular',
+            tag_category='molnix_region')
+        surge_alert_1 = SurgeAlert.objects.create(
+            message='CEA Coordinator, Floods, Atlantis', atype=6, category=2, molnix_status='active', country=country_1)
+        surge_alert_1.molnix_tags.set([molnix_tag_1, molnix_tag_2])
+        surge_alert_2 = SurgeAlert.objects.create(
+            message='WASH Coordinator, Earthquake, Neptunus', atype=5, category=1, molnix_status='active', country=country_2)
+        surge_alert_1.molnix_tags.set([molnix_tag_1, molnix_tag_2])
+        surge_alert_2.molnix_tags.set([molnix_tag_1, molnix_tag_3])
```

Comment:
We can also use factoryboy here.

Reply:
I'd keep it this way, due to there is not yet factory in this notifications app and it would be a huge work.
Not mentioning the experience that factoryboy sometimes forces snapshot-test regenerating without any real reason.

Reply:
I meant just the factories, without snapshot-test. I will push this changes.

Reply:
I am not against factories (mainly if it's not me who makes it), but sometimes there is some relevance the "ambiance of real strings" in tests, I mean OP-6700 as molnix-tag instead of jUierwKLJ. It can be easier to understand the purpose of the tests with human-readable strings.